manifest/etcd.yaml

```diff
@@ -0,0 +1,148 @@
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: etcd
+spec:
+  secretName: etcd-secret
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  subject:
+    organizations:
+      - redhat
+  isCA: false
+  privateKey:
+    algorithm: RSA
+    encoding: PKCS1
+    size: 2048
+  usages:
+    - server auth
+    - client auth
+  dnsNames:
+    - etcd
+    - etcd-0
+    - etcd-1
+    - etcd-2
+    - etcd-0.etcd
+    - etcd-1.etcd
+    - etcd-2.etcd
+  ipAddresses:
+    - 0.0.0.0
+  issuerRef:
+    name: kcp
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: etcd-peer
+spec:
+  secretName: etcd-peer-secret
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  subject:
+    organizations:
+      - redhat
+  isCA: false
+  privateKey:
+    algorithm: RSA
+    encoding: PKCS1
+    size: 2048
+  usages:
+    - server auth
+    - client auth
+  dnsNames:
+    - etcd-0.etcd
+    - etcd-1.etcd
+    - etcd-2.etcd
+    - etcd-0
+    - etcd-1
+    - etcd-2
+  ipAddresses:
+    - 0.0.0.0
+  issuerRef:
+    name: kcp
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: etcd
+spec:
+  clusterIP: None
+  ports:
+    - port: 2379
+      name: client
+    - port: 2380
+      name: peer
```

Comment:
Do we need to specify the protocol for the ports, like you did for the kcp service?

Reply:
It's okay since the default protocol for services is TCP.